app/services/induction/delete_record.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+class Induction::DeleteRecord < BaseService
+  class DeleteInductionRecordRestrictionError < StandardError; end
+
+  def call
+    ActiveRecord::Base.transaction do
+      check_record_is_deletable!
+
+      update_previous_record
+      delete_induction_record
+    end
+  end
+
+private
+
+  attr_reader :induction_record
+
+  def initialize(induction_record:)
+    @induction_record = induction_record
+  end
+
+  def check_record_is_deletable!
+    raise DeleteInductionRecordRestrictionError, "Cannot delete record because it is active" if active?
+    raise DeleteInductionRecordRestrictionError, "Cannot delete record because it is not in the middle of the induction records history" unless middle_of_history?
+    raise DeleteInductionRecordRestrictionError, "Cannot delete record because the school transfer flag does not matches the previous record" if transfer_flag_changed?
+    raise DeleteInductionRecordRestrictionError, "Cannot delete record because the training status does not matches the previous record" if training_status_changed?
+    raise DeleteInductionRecordRestrictionError, "Cannot delete record because the mentor does not matches the previous record" if mentor_changed?
+  end
```

Comment:
what about if the induction_programme changed?

Reply:
or the schedule?

Reply:
or the preferred_identity?

Reply:
@ltello the `check_record_is_deletable!` has just a few extra safeguards that we thought are the most important ones, but we don't expect any of the records we are going to delete to have any of those changes really.

The bug created dup of the completed records, and such records only set the `induction_status = completed` and `mentor_profile = nil`.

Reply:
You can see the query in the rake task is only picking records created within the time window the bug was running and were "complete" records historically that changed to "change".

```
bug_introduced_at = "2023-09-04"
job_running_window = { from: "00:00:00", to: "11:00:00" }

dup_induction_records = InductionRecord
      .joins(:participant_profile)
      .joins(:versions)
      .where.not(participant_profile: { induction_completion_date: nil })
      .where(induction_status: :changed, school_transfer: false)
      .where("induction_records.created_at >= ?", bug_introduced_at)
      .where("induction_records.created_at::time >= :from AND induction_records.created_at::time <= :to ", job_running_window)
      .where("versions.object_changes ->> 'induction_status' ILIKE '[%completed%changed%]'")
      .where("versions.whodunnit IS NULL")
```